### PR TITLE
fix default store destroy...

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**
modify store specs

### CODE REVIEW

File: core/app/models/spree/store.rb [hunk 1]
```diff
@@ -21,28 +22,37 @@ def self.current(domain = nil)
...
     def self.default
       Rails.cache.fetch("default_store") do
-        where(default: true).first_or_initialize
+        find_or_initialize_by(default: true)
       end
     end
 
+    def self.no_default?
+      where(default: true).count.zero?
+    end
```
Comment (jitendra) [inline core/app/models/spree/store.rb]: Please create a scope non_default -> where.not(default: true) and add method has_default?

File: core/app/models/spree/store.rb [hunk 2]
```diff
@@ -21,28 +22,37 @@ def self.current(domain = nil)
 
     def self.default
       Rails.cache.fetch("default_store") do
-        where(default: true).first_or_initialize
+        find_or_initialize_by(default: true)
       end
     end
 
+    def self.no_default?
+      where(default: true).count.zero?
```
Comment (jitendra) [inline core/app/models/spree/store.rb]: please rename the method to has_default?

File: core/spec/models/spree/store_spec.rb [hunk 3]
```diff
@@ -1,24 +1,48 @@
 require 'spec_helper'
 
 describe Spree::Store, :type => :model do
+  # there is a default store created with the test_app rake task.
+  let(:store_1) { Spree::Store.first || create(:store) }
```
Comment (jitendra) [inline core/spec/models/spree/store_spec.rb]: I think we should replace Spree::Store.first with Spree::Store.where(default: true).first

File: core/spec/models/spree/store_spec.rb [hunk 4]
```diff
@@ -32,47 +56,82 @@
...
+    context 'when default store does not exist' do
+      before { Spree::Store.destroy_all }
+      it { expect(Spree::Store.no_default?).to eq(true) }
+    end
+  end
+
+  describe '#remove_previous_default' do
+    before do
+      store_1.update_column(:default, true)
+      store_2.update_column(:default, true)
```
Comment (jitendra) [inline core/spec/models/spree/store_spec.rb]: please remove this line if its not needed

**Comment by aishmita-aggarwal** [comment]
@jitendra Done with changes. Please check.

**Comment by jitendra** [comment]
:+1: please go ahead.

**Pull request closed without merging**